### PR TITLE
[RS][DDCE-911] Added organisation contact details page.

### PR DESCRIPTION
**Pull request by RafSobol**
# DDCE-(StoryNumberHere)

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/nominees/OrganisationNomineeContactDetailsController.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import pages.nominees.{OrganisationNomineeContactDetailsPage, OrganisationNomineeNamePage}
+import pages.sections.Section9Page
+import play.api.mvc._
+import repositories.UserAnswerRepository
+import views.html.nominees.OrganisationNomineeContactDetailsView
+
+import scala.concurrent.Future
+
+class OrganisationNomineeContactDetailsController @Inject()(
+                                                   val sessionRepository: UserAnswerRepository,
```
Comment (abhijeetardale) [inline app/controllers/nominees/OrganisationNomineeContactDetailsController.scala:35]: 2 space alignment please

File: app/controllers/nominees/OrganisationNomineeNameController.scala [hunk 2]
```diff
@@ -19,33 +19,33 @@ package controllers.nominees
...
-   val sessionRepository: UserAnswerRepository,
-   val navigator: NomineesNavigator,
-   identify: AuthIdentifierAction,
-   getData: UserDataRetrievalAction,
-   requireData: DataRequiredAction,
-   formProvider: WhatIsTheNameOfOrganisationFormProvider,
-   val controllerComponents: MessagesControllerComponents,
-   view: WhatIsTheNameOfTheOrganisationView
+class OrganisationNomineeNameController @Inject()(
+                                                   val sessionRepository: UserAnswerRepository,
```
Comment (abhijeetardale) [inline app/controllers/nominees/OrganisationNomineeNameController.scala:35]: 2 space alignment please

File: test/pages/nominees/OrganisationNomineeContactDetailsPageSpec.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import models.nominees.OrganisationNomineeContactDetails
+import org.scalacheck.Arbitrary
+import pages.behaviours.PageBehaviours
+
+class OrganisationNomineeContactDetailsPageSpec extends PageBehaviours {
+
+  implicit lazy val arbitraryContactDetails: Arbitrary[OrganisationNomineeContactDetails] = Arbitrary {
+    OrganisationNomineeContactDetails("0123123123", "email@email.com")
+  }
+  "OrganisationNomineeContactDetailsPage" must {
```
Comment (abhijeetardale) [inline test/pages/nominees/OrganisationNomineeContactDetailsPageSpec.scala:28]: one space please

File: app/views/nominees/OrganisationNomineeContactDetailsView.scala.html [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+   @inputText(
+    form,
+    id = "phoneNumber",
+    name = "phoneNumber",
+    label = "organisationContactDetails.phoneNumber.label",
+    isPageHeading = false,
+    classes = Some("govuk-input govuk-input--width-20")
+   )
+   <br>
```
Comment (abhijeetardale) [inline app/views/nominees/OrganisationNomineeContactDetailsView.scala.html:49]: is br required?

File: conf/messages.en [hunk 5]
```diff
@@ -1323,14 +1323,28 @@ individualNomineesNino.error.format = Enter a National Insurance number in the c
...
+organisationContactDetails.phoneNumber.label = Phone number
+organisationContactDetails.phoneNumber.error.format = Enter the organisation’s main phone number, like 01632 960 001, 07700 900 982 or +44 0808 157 0192
+organisationContactDetails.phoneNumber.error.required = Enter the organisation’s phone number
+organisationContactDetails.email.label = Email address
+organisationContactDetails.email.error.format = Enter your charity’s email address in the correct format, like name@example.com
+organisationContactDetails.email.error.length = The organisation’s email address must be 160 characters or fewer.
+organisationContactDetails.email.error.required = Enter the organisation’s email address
+organisationContactDetails.phoneNumber.checkYourAnswersLabel = Phone number
+organisationContactDetails.email.checkYourAnswersLabel = Email address
+
```
Comment (abhijeetardale) [inline conf/messages.en:1347]: could you please check message files CY and EN both should be in sync, line number changes not cosistent, missing something in en file.

**Review by abhijeetardale — CHANGES_REQUESTED** [review]

File: app/views/nominees/OrganisationNomineeContactDetailsView.scala.html (hunk 4 above)
Comment (RafSobol) [reply app/views/nominees/OrganisationNomineeContactDetailsView.scala.html:49]: it's required to make it look exactly the same as the prototype (which contains the <br> there) - I can either replace it with some custom css, ignore it (so the two fields will be a little closer together), or keep it as a <br>

File: test/navigation/NomineesNavigatorSpec.scala [hunk 6]
```diff
@@ -303,16 +318,30 @@ class NomineesNavigatorSpec extends SpecBase {
...
             routes.SessionExpiredController.onPageLoad()
         }
 
         "go to the summary page when continue button is clicked" in {
-          navigator.nextPage(WhatIsTheNameOfOrganisationPage, CheckMode,
-            emptyUserAnswers.set(WhatIsTheNameOfOrganisationPage, "abc").success.value) mustBe
+          navigator.nextPage(OrganisationNomineeContactDetailsPage, NormalMode,
+            emptyUserAnswers.set(OrganisationNomineeContactDetailsPage, OrganisationNomineeContactDetails("0123123123", "test@email.com")).success.value) mustBe
             routes.DeadEndController.onPageLoad() // TODO when next page is ready
         }
```
Comment (abhijeetardale) [inline test/navigation/NomineesNavigatorSpec.scala:346]: unit tests are missing for OrganisationNomineeContactDetailsPage for check mode

**Review by abhijeetardale — CHANGES_REQUESTED** [review]

**Review by abhijeetardale — APPROVED** [review]

**Pull request merged**